src/components/Calculator.js

```diff
@@ -1,30 +1,46 @@
 import React from 'react';
+import calculate from './logic/calculate.js';
 
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(e) {
+    this.setState((state) => (calculate(state, e.target.textContent)));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className='calculator-container'>
-        <div className='number-field'>{this.props.number}</div>
+        <div className='number-field'>{ total } { operation } { next }</div>
         <div className='numbers-container'>
-          <button className='calc-button'>AC</button>
-          <button className='calc-button'>+/-</button>
-          <button className='calc-button'>%</button>
-          <button className='calc-button orange-btn'>÷</button>
-          <button className='calc-button'>7</button>
-          <button className='calc-button'>8</button>
-          <button className='calc-button'>9</button>
-          <button className='calc-button orange-btn'>X</button>
-          <button className='calc-button'>4</button>
-          <button className='calc-button'>5</button>
-          <button className='calc-button'>6</button>
-          <button className='calc-button orange-btn'>-</button>
-          <button className='calc-button'>1</button>
-          <button className='calc-button'>2</button>
-          <button className='calc-button'>3</button>
-          <button className='calc-button orange-btn'>+</button>
-          <button className='calc-button zero-btn'>0</button>
-          <button className='calc-button'>.</button>
-          <button className='calc-button orange-btn'>=</button>
+          <button className='calc-button' onClick={this.clickHandler}>AC</button>
+          <button className='calc-button' onClick={this.clickHandler}>+/-</button>
+          <button className='calc-button' onClick={this.clickHandler}>%</button>
+          <button className='calc-button orange-btn' onClick={this.clickHandler}>÷</button>
+          <button className='calc-button' onClick={this.clickHandler}>7</button>
+          <button className='calc-button' onClick={this.clickHandler}>8</button>
+          <button className='calc-button' onClick={this.clickHandler}>9</button>
+          <button className='calc-button orange-btn' onClick={this.clickHandler}>X</button>
```

Comment:
Every time I use the multiplication operator (`x`) the calculator component disappears on the page.

- Kindly look into this to make sure that all operations work smoothly

Otherwise, everything else works perfectly. Great job!!! 👏🏾